base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java

```diff
@@ -496,15 +559,23 @@ private void retrieveCapability(final SystemCapabilityType systemCapabilityType,
 		}
 		final GetSystemCapability request = new GetSystemCapability();
 		request.setSystemCapabilityType(systemCapabilityType);
+		request.setSubscribe(subscribe != null ? subscribe : isSubscribedToSystemCapability(systemCapabilityType)); // If subscribe is null, then don't change the current subscription status
 		request.setOnRPCResponseListener(new OnRPCResponseListener() {
 			@Override
 			public void onResponse(int correlationId, RPCResponse response) {
-				if(response.getSuccess()){
+				if (response.getSuccess()) {
 					Object retrievedCapability = ((GetSystemCapabilityResponse) response).getSystemCapability().getCapabilityForType(systemCapabilityType);
 					cachedSystemCapabilities.put(systemCapabilityType, retrievedCapability);
-					if(scListener!=null){scListener.onCapabilityRetrieved(retrievedCapability);	}
-				}else{
-					if(scListener!=null){scListener.onError(response.getInfo());}
+					if (scListener != null) {
+						scListener.onCapabilityRetrieved(retrievedCapability);
```

Comment:
Would this cause the listener to receive this information twice, once from here and once from `notifyListeners`

Reply:
I don't think so because `notifyListeners` should not get called in that case.

Reply:
I think you are right. I will test that.